src/sql/workbench/services/executionPlan/common/executionPlanService.ts

```diff
@@ -27,25 +27,49 @@ export class ExecutionPlanService implements IExecutionPlanService {
 		this._providerRegisterEvent = this._onProviderRegister.event;
 	}
 
-	private async ensureCapabilitiesRegistered(): Promise<void> {
-		let providers = Object.keys(this._capabilitiesService.providers);
-		if (!providers) {
-			await new Promise<void>(resolve => {
-				this._capabilitiesService.onCapabilitiesRegistered(e => {
+	/**
+	 * This ensures that the capabilities service has registered the providers to handle execution plan requests for the given file format.
+	 * @param fileExtension Execution plan file format
+	 */
+	public async ensureFileExtensionHandlerRegistered(fileExtension: string) {
+		for (let provider in Object.keys(this._capabilitiesService.providers)) {
```

Comment:
nit: It would be good to clarify this as `providerId` - it's not the actual provider itself. (same below in _runAction)